unison-src/transcripts/cycle-update-3.output.md

```diff
@@ -0,0 +1,66 @@
+Update a member of a cycle with a type-changing update, thus severing the cycle.
+
+```unison
+ping : 'Nat
+ping _ = !pong + 1
+
+pong : 'Nat
+pong _ = !ping + 2
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      ping : 'Nat
+      pong : 'Nat
+
+```
+```ucm
+.> add
+
+  ⍟ I've added these definitions:
+  
+    ping : 'Nat
+    pong : 'Nat
+
+```
+```unison
+ping : Nat
+ping = 3
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These names already exist. You can `update` them to your
+      new definition:
+    
+      ping : Nat
+
+```
+```ucm
+.> update
+
+  ⍟ I've updated these names to your new definition:
+  
+    ping : Nat
+
+.> view ping pong
+
+  ping : Nat
+  ping = 3
+  
+  pong : 'Nat
+  pong _ =
+    use Nat +
+    !#4t465jk908.1 + 2
```

Comment:
This is probably a whole different issue, but do we do anything to warn the user that this is happening? Or is `pong` at least added to the update frontier or something?

Even in that case I don't see any warning being output by UCM; we really need to strive to avoid any possible workflow that orphans references like this :'( 

Any ideas of something we could do to improve UX here?

Reply:
Yeah, this case actually isn't different from a normal type-changing update, without complications due to cycles. In both cases, ucm allows the update, then you can run `todo` and see what references the old thing. Agree that it's probably useful to know if one would cause a `todo` entry before applying an update.

As of https://github.com/unisonweb/unison/pull/3400 at least you'll see the busted thing.